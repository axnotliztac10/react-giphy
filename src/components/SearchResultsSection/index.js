import React from 'react';

import GiphyCard from 'components/GiphyCard';
import { Text, Box, Flex, Link } from 'rebass';

export default class SearchResultsSection extends React.PureComponent {
  renderResult = (result, index) => {
    const { onAddRemoveFavorite } = this.props;

    return (
      <GiphyCard
        giphy={result}
        key= {index}
        onAddRemoveFavorite={onAddRemoveFavorite}
      />
    );
  }

  render() {
    const { results, onLoadMore } = this.props;

    if (!results.length) {
      return null;
    }

    return (
      <Box my={4}>
        <Text textAlign="center" fontSize={22}>Results:</Text>
        <Flex
          flexDirection="row"
          justifyContent="center"
          alignItems="flex-start"
          flexWrap="wrap"
        >
          {results.map(this.renderResult)}
          <Box width={1}>
            <Link fontSize={30} onClick={() => onLoadMore()}>Load more</Link>
          </Box>
        </Flex>
      </Box>
    );
  }
}
