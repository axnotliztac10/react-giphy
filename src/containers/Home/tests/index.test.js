import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import SearchResultsSection from 'components/SearchResultsSection';

import { HomeÂ } from '../index';

describe('<Home />', () => {
  const wrapper = shallow(
    <Home
      onSearch={() => {}}
      trendingList={[]}
      searchList={[]}
      onAddRemoveFavorite={() => {}}
      searchPagination={{}}
    />
  );

  it('should render correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Should trigger prop', () => {
    wrapper.find(SearchResultsSection).prop('onLoadMore')();
  });
});
