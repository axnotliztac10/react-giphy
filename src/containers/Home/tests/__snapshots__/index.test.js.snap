// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<Layout
  heading="React Challenge"
>
  <TrendingSection
    onAddRemoveFavorite={[Function]}
    trendings={Array []}
  />
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="search"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    onSearch={[Function]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
  <SearchResultsSection
    onAddRemoveFavorite={[Function]}
    onLoadMore={[Function]}
    results={Array []}
  />
</Layout>
`;
